src/Components/CardList.tsx

```diff
@@ -80,14 +90,88 @@ export const CardList = ({
     if (item?.format && formats.includes(item?.format)) return true
   }
 
+  function genderFilter(
+    item: ServicePreview,
+    genderInput: Array<string> | undefined
+  ) {
+    // If no preference is set for user or service, return everything
+    if (genderInput?.length === 0 || item?.gender?.length === 0) {
+      return true
+    }
+
+    genderInput?.filter((gen) => {
+      const gender = gen as Gender
```

Comment:
Using `as` for internal code sort of says to me that something is wrong... in this case I've tracked it back to the way I set up the quiz context. There isn't type validation for `CategoryList` - it just takes a generic string. If I had made it something like `CategoryList<Gender>`, then here we would have been able to have:

```ts
genderInput: Array<Gender> | undefined
```

and we wouldn't need the `as` conversion.

Anyway, I didn't do any of this when I made `QuizContext`, so I guess this is fine for now.

Reply:
I'm not sure they were defined when you made the `QuizContext` 🙂 

I thought about rummaging in here but the PR was already Quite Large. It would be cool if the checkbox inputs and types came from the same list. I think we could unify the type and the values with something like:

const gender = ['men', 'women', 'non-binary', 'transgender' ] as const;
type Gender = typeof gender[number];

Reply:
You can do that? That's really cool.

Reply:
I'll do it now in a separate commit so that there isn't another large diff. It will ease things like having toLower() stuff so that it isn't slightly different.